src/snowflake/snowpark/column.py

```diff
@@ -990,12 +994,14 @@ def _create_ast(
         ast = proto.Expr()
         if property is not None:
             prop_ast = getattr(ast, property)
+            fill_src_position(prop_ast.src)
             for attr, value in assign_fields.items():
                 setattr_if_not_none(prop_ast, attr, value)
             for attr, value in assign_opt_fields.items():
                 setattr_if_not_none(getattr(prop_ast, attr), "value", value)
             for attr, messg in copy_messages.items():
-                getattr(prop_ast, attr).CopyFrom(messg)
+                if messg is not None:
```

Comment:
This error here actually motivated me to look into IR for functions right now :) Not sure we should merge it, I think I'd prefer to throw a NotImplementedError here with a good message which API is responsible for this. Do you think we can make an edit for this?

Reply:
I tried adding some logic which should be able to raise an error which is granular to the line level where the usage of a Snowpark function API that does not perform AST logging is detected. Would this be sufficient?

It doesn't seem straightforward to get the exact API that doesn't log its AST.

Reply:
I'll leave an example of the error message below as well with the code snippet
```python
from snowflake.snowpark import Session
from snowflake.snowpark.functions import mean, stddev, col

session = Session.builder.create()
df = session.table("test_table")

result = df.select((mean(df["STR"]) - 2 * stddev(df["STR"]) < col('STR')) & (col('STR') < mean(df["STR"]) + 2 * stddev(df["STR"])))
print(result.queries)
```

```
NotImplementedError: Calling Column API "__rmul__" which supports AST logging, from File "/Users/heshah/hemit/scratch.py", line 7, potentially in <module>
        result = df.select((mean(df["STR"]) - 2 * stddev(df["STR"]) < col('STR')) & (col('STR') < mean(df["STR"]) + 2 * stddev(df["STR"])))
        An API from snowflake.snowpark.functions used above has not yet implemented AST logging.
```